codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/middleware/MiddlewareBuilder.java

```diff
@@ -147,17 +146,19 @@ private void resolve(Middleware middleware, Set<Middleware> resolved, Set<Middle
         if (!resolved.contains(middleware)) {
             visiting.add(middleware);
             if (middleware.getRelative().isPresent()) {
-                Middleware relativeTo = Objects.requireNonNull(
-                        klassToMiddlewareMap.get(middleware.getRelative().get().getTo()),
-                        middleware.getKlass() + " relative references a middleware class ("
-                                + middleware.getRelative().get().getTo() + ") that is not available in the stack.");
-                //recursively resolve the relative middleware
-                resolve(relativeTo, resolved, visiting, order, klassToMiddlewareMap);
-                // the order of relativeTo should now be set
-                if (middleware.getRelative().get().getType().equals(Middleware.Relative.Type.BEFORE)) {
-                    order.put(middleware, order.get(relativeTo) - 1);
+                Middleware.Relative relative = middleware.getRelative().get();
+                Middleware relativeTo = klassToMiddlewareMap.get(relative.getTo());
+                if (relativeTo != null) {
+                    // recursively resolve the relative middleware
+                    resolve(relativeTo, resolved, visiting, order, klassToMiddlewareMap);
+                    // the order of relativeTo should now be set
+                    if (relative.getType().equals(Middleware.Relative.Type.BEFORE)) {
+                        order.put(middleware, order.get(relativeTo) - 1);
+                    } else {
+                        order.put(middleware, order.get(relativeTo) + 1);
+                    }
                 } else {
-                    order.put(middleware, order.get(relativeTo) + 1);
+                    resolveNullRelativeTo(relative, middleware, order);
```

Comment:
Rather than "Null" in the name, maybe "Missing"

Reply:
Done. Thank you!